src/runtime/executable.jl

```diff
@@ -70,10 +69,55 @@ end
 
 function get_global(exe::ROCExecutable, symbol::Symbol)
     @assert symbol in keys(exe.globals) "No such global in executable: $symbol"
-    return exe.globals[symbol]::Mem.Buffer
+    return exe.globals[symbol]
 end
 
 function getinfo(exsym::HSA.ExecutableSymbol, attribute::HSA.ExecutableSymbolInfo,
                  value::Union{Vector,Base.RefValue,String})
     HSA.executable_symbol_get_info(exsym, attribute, value)
 end
+
+has_exception(e::ROCExecutable) = haskey(e.globals, :__global_exception_flag)
+
+function get_exception(
+    exe::ROCExecutable; check_exceptions::Bool = true, signal_handle::UInt64,
+)
+    has_exception(exe) || return nothing
+
+    # Check if any wavefront for this kernel threw an exception
+    ex_flag = get_global(exe, :__global_exception_flag)
+    ex_flag_ptr = Base.unsafe_convert(Ptr{Int64}, ex_flag)
+    ex_flag_value = Base.unsafe_load(ex_flag_ptr)
+    ex_flag_value == 0 && return nothing
+
+    ex_string = nothing
+    fetch_ex_strings =
+        check_exceptions && haskey(exe.globals, :__global_exception_ring)
+
+    if fetch_ex_strings
+        ex_strings = String[]
+        # Check for and collect any exceptions, and clear their slots
+        ex_ring = get_global(exe, :__global_exception_ring)
+        ex_ring_ptr_ptr = Base.unsafe_convert(
+            Ptr{Ptr{AMDGPU.Device.ExceptionEntry}}, ex_ring)
+        ex_ring_ptr = unsafe_load(ex_ring_ptr_ptr)
+
+        while (ex_ring_value = unsafe_load(ex_ring_ptr)).kern != 1
+            if ex_ring_value.kern == signal_handle
```

Comment:
```suggestion
            if ex_ring_value.kern == signal_handle && ex_ring_value.ptr != C_NULL
```

I seem to recall that this is sometimes necessary, although that's probably a bug.